app-backup/dar: fix some bugs

ceamac opened the pull request:
USE flags "static" and "static-libs" are already masked for almost 6 months now.  They were blocking adding the "rsync" USE flag because `net-libs/librsync` does not support "static-libs" currently.

The bugs are:
- https://bugs.gentoo.org/739698
- https://bugs.gentoo.org/729150
- https://bugs.gentoo.org/767454
- https://bugs.gentoo.org/686278

thesamesam reviewed (changes requested):
Good work. Can you split 'app-backup/dar: add support for threads' into two commits, one which adds the new package, and another which adds the support to dar?

ceamac commented:
Thank you.  I split it into two commits (that's how I wrote them at the beginning, but I squashed them to have one commit per bug).

About the remaining bug in `dar`:
- https://bugs.gentoo.org/771696 - this one was reported by me and was caused by linking with gold, ld.bfd solved the issue.  I grepped portage for ld.bfd and haven't found ebuilds forcing ld.bfd, except a compiler and the kernel, so I think this bug can be closed, not fixed.

thesamesam commented on dev-libs/libthreadar/libthreadar-1.3.0.ebuild:
```diff
@@ -0,0 +1,42 @@
+
+LICENSE="LGPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~sparc ~x86 ~amd64-linux"
```
We need to drop to just ~amd64 and request rekeywording of the new version of dar.

thesamesam commented on app-backup/dar/metadata.xml:
```diff
@@ -15,6 +22,7 @@ operations, including with ssh.
 		by 64 bit integers</flag>
 		<flag name="gcrypt">Enables strong encryption support</flag>
 		<flag name="gpg">Enables usage of asymetric crypto algorithms</flag>
+		<flag name="rsync">Enables linking with net-libs/librsync for delta compression support</flag>
```
```suggestion
		<flag name="rsync">Enables linking with <pkg>net-libs/librsync</pkg> for delta compression support</flag>
```
(I'll do it)

thesamesam commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+	# Bug 103741
+	filter-flags -fomit-frame-pointer
+
+	myconf+=( --disable-dar-static --disable-static )
```
I don't think this belongs on its own.

ceamac replied:
The ebuild for 2.6.13 had this:
```
	if ! use static ; then
		myconf+=( --disable-dar-static )
		if ! use static-libs ; then
			myconf+=( --disable-static )
		fi
	fi
```
So when I removed use static and use static-libs, I simply resolved the if.  Maybe put it on two lines?  I think the way configure is written, all not disabled is enabled by default

thesamesam commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+		$(usex dar32 --enable-mode=32 '')
+		$(usex dar64 --enable-mode=64 '')
+		$(usex doc '' --disable-build-html)
+		#$(usex examples --enable-examples '')
```
Why?

ceamac replied:
I checked.  `--disable-examples` stops with an error, `--enable-examples` does nothing (same files are installed in docs).  So I removed it.

thesamesam commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+		sys-devel/gettext
+		virtual/libintl
+	)
+	doc? ( app-doc/doxygen )
```
I'll sort BDEPEND.

thesamesam commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+
+RESTRICT="test" # need to be run as root
+
+RDEPEND="
```
I'll sort RDEPEND.

thesamesam commented on dev-libs/libthreadar/libthreadar-1.3.0.ebuild:
```diff
@@ -0,0 +1,42 @@
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~sparc ~x86 ~amd64-linux"
+
+DEPEND=""
```
Not needed - see [here](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-620007.3)

dwfreed commented on dev-libs/libthreadar/metadata.xml:
```diff
@@ -0,0 +1,19 @@
+	</maintainer>
+	<longdescription>
+Libthreadar is a C++ library providing an abstracted set of C++ *classes* to manipulate threads in a very simple and efficient way from your C++ code.
+It also handles exceptions thrown from a thread and propagated to another one, when the later is calling the thread::join() method. This let one manage exceptions as simply as it is in C++ single threaded context.
```
```suggestion
It also handles exceptions thrown from a thread and propagated to another one, when the latter is calling the thread::join() method. This lets one manage exceptions as simply as it is in C++ single threaded context.
```

dwfreed commented on app-backup/dar/metadata.xml:
```diff
@@ -15,6 +22,7 @@ operations, including with ssh.
 		by 64 bit integers</flag>
 		<flag name="gcrypt">Enables strong encryption support</flag>
 		<flag name="gpg">Enables usage of asymetric crypto algorithms</flag>
```
```suggestion
		<flag name="gpg">Enables usage of asymmetric encryption algorithms</flag>
```

juippis commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+	local DOCS=( AUTHORS ChangeLog NEWS README THANKS TODO )
+	einstalldocs
+
+	find "${ED}" -name '*.la' -delete || die
+	find "${ED}" -name '*.a' -delete || die
```
```suggestion
	find "${ED}" \( -name "*.a" -o -name "*.la" \) -delete || die
```

juippis commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+	find "${ED}" -name '*.a' -delete || die
+
+	# Bug 729150
+	(cd "${ED}/usr/share/doc/${PF}/html/samples" && mkdir MyBackup && tar zxf MyBackup.sh.tar.gz -C MyBackup && rm MyBackup.sh.tar.gz) || die
```
What's up with this monster? How can you tell which part fails now? Actually can you walk me through what's happening here, because there's most likely a better way to deal with it. 

If you look at *Makefile.in* file and grep for *gzip* you most likely start to see where this is coming from. We'd prefer a build-system patch. I believe simply emptying out `DIST_TARGETS` will fix it, but please do test with 
https://wiki.gentoo.org/wiki/Package_testing#make.conf

ceamac replied:
It doesn't come from there.  There is a tar.gz file included in the distribution, please check `dar-2.6.13/doc/samples/MyBackup.sh.tar.gz`, it is not built by the build system, only copied as is.  `DIST_TARGETS` applies to `make dist`, not to `make install`.

I unpacked it because the bug is an automated check.  Other ways to fix it would be:
- patch the build system to unpack it at `make install` time
- or simply remove it from installation

How do you suggest I fix this bug?

juippis replied:
Hmm I see. Can you try calling `docompress -x` on it in `src_install`, which ignores the package-manager compression for it?

Is the package any relevant for the user, does it make sense installing it?

ceamac replied:
After a bit of searching it seems unpacking it is not a solution, because that directory contains this page: http://dar.linux.free.fr/doc/samples/index.html

Unpacking `MyBackup.sh.tar.gz` would break the link in `index.html` anyway.  So if I remove it and someone really needs it, they can download it.  Is this acceptable?

ceamac replied:
Also I'll check if this directory is related to that "examples" USE flag from earlier.

juippis replied:
Yes please, sounds good.

juippis commented on app-backup/dar/dar-2.6.13-r1.ebuild:
```diff
@@ -0,0 +1,90 @@
+src_install() {
+	emake DESTDIR="${D}" pkgdatadir="${EPREFIX}"/usr/share/doc/${PF}/html install
+
+	local DOCS=( AUTHORS ChangeLog NEWS README THANKS TODO )
```
Please move this to global scope, between `REQUIRED_USE` and `src_configure` for example.

ceamac commented:
I think (I hope) I've solved everything.
I masked "threads" USE flag on *ppc*, *sparc* and *x86*, and I've got a conflict with master after pushing.  I hope I fixed it ok.
Please check and review again.

juippis commented on dev-libs/libthreadar/libthreadar-1.3.0.ebuild:
```diff
@@ -0,0 +1,37 @@
+	# Do _not_ use $(use_enable) until you have verified that the
+	# logic has been fixed by upstream.
+
+	local myconf=( --disable-static )
+
+	econf "${myconf[@]}"
```
You could just call `econf --disable-static` here, but I don't mind this.

juippis reviewed (approved):
Thanks! 

Regarding the co-authors, I had to read https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin if it's legal to include them. I'd say yes for now.
